test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+
+    # Reboot the MicroShift system vm after rollback
+    Reboot MicroShift System
+    Sleep   200s
+   
+    # Verify microshift cluster work fine after rollback
+    Get NodeName
+
+*** Keywords ***
+Generate New IP
+    ${hostip}     ${rc1}=    Execute Command    microshift show-config | grep -i "nodeIP:" | awk '{print $2}'    sudo=True   return_stdout=True    return_rc=True
+    Should Be Equal As Integers    ${rc1}    0
+    ${gateway}    ${rc2}=    Execute Command    ip route show | grep -i ${hostip} | grep -i default | awk '{print $3}'    return_stdout=True    return_rc=True
+    Should Be Equal As Integers    ${rc2}    0
+    ${device}     ${rc3}=    Execute Command    ip route show | grep -i ${hostip} | grep -i default | awk '{print $5}'    return_stdout=True    return_rc=True
+    Should Be Equal As Integers    ${rc3}    0
+
+    FOR    ${last_digit}    IN RANGE    2    255
+        ${digit}    Evaluate    int($hostip.split('.')[-1])
+        ${new_host_ip}    Evaluate    '${hostip}'.rsplit('.', 1)[0] + '.' + str(${last_digit})
+        Log    Attempt ${last_digit}: Generated IP: ${new_host_ip}
+        ${ping_output}     ${rc4}=    Execute Command    ping -c 1 ${new_host_ip}     return_stdout=True    return_rc=True
+        Run Keyword If     ${rc4} > 0    Exit For Loop
+    END
+
+    Log    Original Host IP: ${hostip}
+    Log    New Host IP: ${new_host_ip}
+    Log    Device Name: ${device}
+
+    # Add one more check to ensure the generated IP is not the same as the current IP
+    Should Not Be Equal    ${new_host_ip}    ${hostip}
+
+    # Set the global variables
+    Set Global Variable    ${NEW_HOST_IP}    ${new_host_ip}
+    Set Global Variable    ${GATEWAY}       ${gateway}
+    Set Global Variable    ${DEVICE}        ${device}
+
+    # Set the global variable for nmconnection_path
+    Set Global Variable    ${NMCONNECTION_PATH}    /etc/NetworkManager/system-connections/${device}.nmconnection
+
+Save Backup Default File
+    [Arguments]    ${file_path}
+    [Documentation]    Takes a backup of the given file and returns the path to the backup file.
+    ...    This keyword is meant to be used from a Setup step.
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp ${file_path} ${file_path}.backup
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Restore Backup File
+    [Arguments]    ${backup_file}
+    [Documentation]    Restores the original file from the backup file.
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp ${backup_file}.backup ${backup_file}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Setup Suite
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig    # for readiness checks
+
+Rollback Configuration
+    [Arguments]    ${backup_file}
+    [Documentation]    Restore the original configuration from the backup file
```

Comment:
This is a very thin wrapper around "Restore Backup File". What file is it restoring? Can the caller just use "Restore Backup File" directly?

Reply:
Done